variables.tf

```diff
@@ -338,3 +338,18 @@ variable "cluster_encryption_config" {
   }))
   default = []
 }
+
+variable "fargate_profiles" {
+  description = "Fargate profiles"
+  type = map(object({
+    namespace = string
+    labels    = map(string)
+  }))
+  default = {}
+}
+
+variable "create_fargate_pod_execution_role" {
+  description = "Controls if the EKS Fargate pod execution IAM role should be created."
+  type        = bool
+  default     = false
```

Comment:
Could creation of the IAM role be based on the length of `fargate_profiles`? That way this variable could default to true.

And if we're giving the option of not creating the fargate role we should provide the ability to pass in an externally created role ARN.